src/puppetlabs/puppetdb/cli/benchmark.clj

```diff
@@ -271,140 +290,110 @@
-  (activate-logging! options)
   (cond
     (and (contains? options :runinterval)
          (contains? options :nummsgs))
-    (do
-      (log/error "Error: -N/--nummsgs runs immediately and is not compatable with -i/--runinterval")
-      (action-on-error-fn))
-
-    (not-any? #{:nummsgs :runinterval} (keys options))
-    (do
-      (log/error "Error: Either -N/--nummsgs or -i/--runinterval is required.")
-      (action-on-error-fn))
+    (utils/throw+-cli-error! "Error: -N/--nummsgs runs immediately and is not compatable with -i/--runinterval")
+    
+    (and (not (contains? options :runinterval))
+         (not (contains? options :nummsgs)))
+    (utils/throw+-cli-error! "Error: Either -N/--nummsgs or -i/--runinterval is required.")
 
     (and (contains? options :archive)
          (some #{:reports :catalogs :facts} (keys options)))
-    (do
-      (log/error "Error: -A/--archive is incompatible with -F/--facts, -C/--catalogs, -R/--reports")
-      (action-on-error-fn))
+    (utils/throw+-cli-error! "Error: -A/--archive is incompatible with -F/--facts, -C/--catalogs, -R/--reports")
 
     :else options))
 
-(defn default-options
-  [options]
-  (assoc options
-         :facts "puppetlabs/puppetdb/benchmark/samples/facts"
-         :reports "puppetlabs/puppetdb/benchmark/samples/reports"
-         :catalogs "puppetlabs/puppetdb/benchmark/samples/catalogs"
-         :from-cp? true))
-
 (defn- validate-cli!
   [args]
-  (try+
-    (let [options (-> (ks/cli! args supported-cli-options required-cli-options)
-                      first
-                      ;; We load the config for logging and for our main-
-                      (update :config config/load-config)
-                      (validate-options #(System/exit 1)))]
-      (if (empty? (select-keys options [:facts :reports :catalogs]))
-        (default-options options)
-        options))
-
-    (catch map? m
-      (println (:message m))
-      (case (:type m)
-        :puppetlabs.kitchensink.core/cli-error (System/exit 1)
-        :puppetlabs.kitchensink.core/cli-help (System/exit 0)))))
+  (let [specs [["-c" "--config CONFIG" "Path to config or conf.d directory (required)"
+                :parse-fn config/load-config]
+               ["-F" "--facts FACTS" "Path to a directory containing sample JSON facts (files must end with .json)"]
+               ["-C" "--catalogs CATALOGS" "Path to a directory containing sample JSON catalogs (files must end with .json)"]
+               ["-R" "--reports REPORTS" "Path to a directory containing sample JSON reports (files must end with .json)"]
+               ["-A" "--archive ARCHIVE" "Path to a PuppetDB export tarball. Incompatible with -C, -F, -R, or -D"]
+               ["-i" "--runinterval RUNINTERVAL" "What runinterval (in minutes) to use during simulation"]
+               [nil "--from-classpath FROMCLASSPATH" "Whether to retrieve files from the classpath"
+                :default true
+                :parse-fn kitchensink/parse-bool]
```

Comment:
won't this effectively be always true now? I wonder if that'll mess up imports with -F, -C, and -R. I don't understand making it a CLI option rather than inferring it from the absence of those three params.